### PR TITLE
[ExpansionPanel] Exposes color property of MaterialSlice

### LINKED ISSUE #32293: ExpansionPanel List Styling Options Limited to cardColor
I have been trying to use flutter's `ExpansionPanelList` widget, however, styling options are quite limited. The widget automatically uses `cardColor` attribute of the current theme and from what I see, the only option to apply custom styles/widgets is to override the `_ExpansionPanelListState`, which I can't because it is private. As per this [S.O comment](https://stackoverflow.com/questions/56030464/flutter-expansion-panel), currently the only solution is to copy the entire `expansion_panel.dart` file and override the widgets's `build` method.

### PR DESCRIPTION
**Pull request by AyushBherwani1998**
## Description

Adds a new `backgroundColor`  property in `EpxansionPanel`. The property uses `MaterialSlice.color` to paint the panel color.

## Related Issues
Fixes #32293

## Tests

I added the following tests:

- Added test to verify the `panelColor` is propagated properly to MaterialSlice.

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by AyushBherwani1998** [comment]
Oh no, I don't know how the stale commit is present in the branch. Let me know if I should send a fresh PR.

File: packages/flutter/test/material/expansion_panel_test.dart [hunk 1]
```diff
@@ -1458,4 +1458,40 @@ void main() {
       ' possible elevation values.'
     ));
   });
+
+  testWidgets('ExpansionPanel.panelColor test', (WidgetTester tester) async {
```
Comment (shihaohong) [inline packages/flutter/test/material/expansion_panel_test.dart:1462]: We should add another similar test for `ExpansionPanelRadio`, just to make sure we don't regress there.

**Review by shihaohong — COMMENTED** [review]
The screenshot looks great! A nit or two and a request for a test for `ExpansionPanelRadio`

File: packages/flutter/test/material/expansion_panel_test.dart [hunk 2]
```diff
@@ -1494,4 +1494,39 @@ void main() {
     expect((mergeableMaterial.children.first as MaterialSlice).color, firstPanelColor);
     expect((mergeableMaterial.children.last as MaterialSlice).color, secondPanelColor);
   });
+
```
Comment (shihaohong) [inline packages/flutter/test/material/expansion_panel_test.dart:1497]: Failing analyzer check:
```suggestion

```

**Review by shihaohong — APPROVED** [review]
LGTM when the bots are happy

File: packages/flutter/lib/src/material/expansion_panel.dart [hunk 3]
```diff
@@ -101,6 +102,10 @@ class ExpansionPanel {
   /// Defaults to false.
   final bool canTapOnHeader;
 
+  /// Defines the color of panel.
+  ///
+  /// Defaults to [ThemeData.cardColor].
+  final Color? panelColor;
```
[the remark this answers is not in the document]
Comment (AyushBherwani1998) [reply packages/flutter/lib/src/material/expansion_panel.dart]: Make sense to me. I have renamed the property.

**Pull request merged**